system-test/logging.ts

```diff
@@ -182,6 +182,37 @@ describe('Logging', () => {
       });
     });
 
+    describe('metadata with uniqueWriterIdentity', () => {
+      it('should set metadata if uniqueWriterIdentity was true', async () => {
+        const sink = logging.sink(generateName());
+        const FILTER = 'severity = ALERT';
+        await sink.create({
+          destination: topic,
+          uniqueWriterIdentity: true,
+        });
+        const metadata = {
+          filter: FILTER,
+          uniqueWriterIdentity: true,
+        };
+        const [apiResponse] = await sink.setMetadata(metadata);
+        assert.strictEqual(apiResponse.filter, FILTER);
+      });
+
+      it('should set uniqueWriterIdentity from false to true', async () => {
+        const sink = logging.sink(generateName());
+        const FILTER = 'severity = ALERT';
+        await sink.create({
+          destination: topic,
+        });
+        const metadata = {
+          filter: FILTER,
+          uniqueWriterIdentity: true,
+        };
+        const [apiResponse] = await sink.setMetadata(metadata);
+        assert.strictEqual(apiResponse.filter, FILTER);
```

Comment:
Not sure how you check that uniqueWriterIdentity was changed from false to true in this case. I see you compare filters and filter is string that doesn't include uniqueWriterIdentity

Reply:
In this test, I only check that users are able to edit filter, given their uniqueWriterId settings. This was the bug that this PR fixes.

You've pinpointed the fundamental issue that uniqueWriterId isn't returned by any API calls at all (which was why I have [this issue](https://github.com/googleapis/nodejs-logging/issues/1035) to help users infer what that boolean was, still waiting on logging-eng to confirm if the inferrence heuristic is reliable)

Reply:
perhaps, worth adding this explanation as comment to the bug or rephrase the name of the test.